sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -792,7 +887,18 @@ class CodegenContext {
       addMutableState(javaType(expr.dataType), value,
         s"$value = ${defaultValue(expr.dataType)};")
 
-      subexprFunctions += s"$fnName($INPUT_ROW);"
+      // Generate the code for this expression tree and wrap it in a function.
```

Comment:
In the original pull-request, since `addMutableState` returned values that were going to be referenced in `fn`, I had to declare the mutable state accessors before the code. It's less important here. The really important part for this pull-request is that the return value of `addNewFunction(fnName, fn)` is given added to `subexprFunctions`.